features/202-session-management/api.md

```diff
@@ -0,0 +1,100 @@
+# Session Management APIs
+
+## SDK API
+```typescript
+interface UserAgent {
+    raw: string;
+    name: string; // browser name or app ID
+    version: string; // browser version or app version
+    os: string;
+    osVersion: string;
+    device: string;
+}
+
+interface Session {
+    id: string;
+    identityID: string;
+    createdAt: Date;
+    lastAccessedAt: Date;
+    createdByIP: string;
+    lastAccessedIP: string;
+    createdByUserAgent: UserAgent;
+    lastAccessedUserAgent: UserAgent;
+    name: string;
+    data: object;
+}
+
+// list all sessions of current user
+function listSessions(): Promise<Session[]>;
+
+// get specific session information for current user
+// for master-key, can use for any user
+function getSession(sessionID: string): Promise<Session>;
+
+// update specific session name for current user
+// for master-key, can use for any user and also update custom attributes
+function updateSession(sessionID: string, name: string | undefined, data?: object): Promise<void>;
```

Comment:
Second thought -- for use case that need to display a custom name in the portal, should it be a session name or device name???

Let's say the case we discussed yesterday, a Skygear APP is to developed with custom Android Devices each name with an ID. The name should be stored in Client Side, and supplied as Devices' information right? hmmm

Reply:
The name displayed in portal would be the session name.

In that case, the developer should mention that the device name would be collected in privacy policy. Then, upon session creation/startup, use `updateSession` API to update the name of session to desired value.

Reply:
So I'm asking it seems better to display the device name instead of session name in Developer Portal?

So if magically someone "stole" a session, and copy it to other devices, one mistake it they may have forgot to supply the same device name and a wrong session is spotted.

Reply:
Device name cannot be obtained reliably, the method to get it is inconsistent on Android platform, and impossible on web platform. It is not useful for a majority of user.

Reply:
I'm talking about a device name that's definable by users here. So instead of a session name, we have a device name that could either be filled in automatically (our client SDK use the best effort to get it, or just the device type), or a device name that can be configured at client side just for display purpose.

Reply:
> I'm talking about a device name that's definable by users here.

I don't understand; session name is already configurable by users?

> a device name that can be configured at client side just for display purpose.

So this device name is a client-side concept only, which SDK would automatically fill in as session name?

Reply:
> I don't understand; session name is already configurable by users?

Yes, but I meant it seems the concept of a user configurable name, is more appropriate to be modelled as a client side configurable name per device, instead of a server side configurable name per session.

The different should be small though, coz unless users do something tricky, the session token is not suppose to be "transfer" to another device.. Got it? and thought?